### ISSUE TITLE
E1020 false positive: Ref to a resource inside UpdatePolicy rejected since 1.56.0 (only parameters/pseudo-parameters allowed)

### ISSUE THREAD

**Issue by roger-zhangg** [aws-cloudformation/cfn-lint#4669]
### CloudFormation Lint Version

1.56.0

### What operating system are you using?

Mac (macOS 26.6.2, Python 3.13.7)

### Describe the bug

Since 1.56.0, `Ref` to a **resource** inside a resource's `UpdatePolicy` is reported as `E1020`. Only pseudo-parameters and template `Parameters` are accepted, so every resource reference is a false positive:

```
repro.yaml:35:9:35:24:E1020:'MyApp' is not one of ['AWS::AccountId', 'AWS::NoValue', 'AWS::NotificationARNs', 'AWS::Partition', 'AWS::Region', 'AWS::StackId', 'AWS::StackName', 'AWS::URLSuffix']
repro.yaml:36:9:36:28:E1020:'MyDeploymentGroup' is not one of ['AWS::AccountId', 'AWS::NoValue', 'AWS::NotificationARNs', 'AWS::Partition', 'AWS::Region', 'AWS::StackId', 'AWS::StackName', 'AWS::URLSuffix']
```

Both `MyApp` and `MyDeploymentGroup` are resources defined in the same template. Refs to `Parameters` in the same position are accepted, which is what pins the cause down: the resource map is empty rather than the whole context being wrong.

**Regression range:** clean on 1.55.1, fails on 1.56.0. Reproduces with freshly updated schemas (`cfn-lint -u --regions us-east-1`).

**Likely cause.** #4641 ("fix: Require object for CreationPolicy and UpdatePolicy", `1a58c98a`, first released in 1.56.0) added `resources={}` to the validator context in `src/cfnlint/rules/resources/updatepolicy/Configuration.py` (rule `E3016`):

```diff
         validator = validator.evolve(
             context=validator.context.evolve(
                 functions=list(FUNCTIONS),
+                resources={},
                 strict_types=False,
             ),
```

Emptying the resource map makes `E1020` reject every resource `Ref` inside `UpdatePolicy`. That addition looks incidental to the PR's stated goal of requiring an object for these attributes. The line is still present on `main`.

**Scope.** Not specific to one policy type — reproduces for both:
- `AWS::Lambda::Alias` → `UpdatePolicy.CodeDeployLambdaAliasUpdate.ApplicationName` / `.DeploymentGroupName`
- `AWS::AutoScaling::AutoScalingGroup` → `UpdatePolicy.AutoScalingRollingUpdate.MaxBatchSize`

For completeness: the same `resources={}` was added to `src/cfnlint/rules/resources/CreationPolicy.py` in the same release, but I could not produce a false positive there — `CreationPolicy` refs I tried stayed clean on both versions. Worth a look while fixing `UpdatePolicy`.

**Real-world impact.** `CodeDeployLambdaAliasUpdate` referencing CodeDeploy resources is exactly what the AWS SAM transform emits for `AutoPublishAlias` + `DeploymentPreference`. AWS SAM's own repo hit this and has pinned cfn-lint to work around it: aws/serverless-application-model#3989. Anyone linting SAM transform output, or hand-writing a Lambda alias with a CodeDeploy deployment preference, sees this.

### Expected behavior

`Ref` inside `UpdatePolicy` should resolve against the template's `Resources` in addition to `Parameters` and pseudo-parameters, as it did through 1.55.1. The template below should lint clean.

CloudFormation itself accepts these references — the resulting templates deploy successfully, and the SAM transform generates them.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  MyApp:
    Type: AWS::CodeDeploy::Application
    Properties:
      ComputePlatform: Lambda

  MyDeploymentGroup:
    Type: AWS::CodeDeploy::DeploymentGroup
    Properties:
      ApplicationName: !Ref MyApp
      ServiceRoleArn: arn:aws:iam::123456789012:role/CodeDeployRole

  MyFunction:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        ZipFile: "def handler(e, c): pass"
      Handler: index.handler
      Role: arn:aws:iam::123456789012:role/LambdaRole
      Runtime: python3.13

  MyVersion:
    Type: AWS::Lambda::Version
    Properties:
      FunctionName: !Ref MyFunction

  MyAlias:
    Type: AWS::Lambda::Alias
    Properties:
      FunctionName: !Ref MyFunction
      FunctionVersion: !GetAtt MyVersion.Version
      Name: live
    UpdatePolicy:
      CodeDeployLambdaAliasUpdate:
        ApplicationName: !Ref MyApp          # E1020 in 1.56.0, clean in 1.55.1
        DeploymentGroupName: !Ref MyDeploymentGroup  # E1020 in 1.56.0, clean in 1.55.1
```

Second case, `AutoScalingRollingUpdate`:

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Resources:
  MyLT:
    Type: AWS::EC2::LaunchTemplate
    Properties:
      LaunchTemplateData:
        ImageId: ami-12345678
  MyASG:
    Type: AWS::AutoScaling::AutoScalingGroup
    Properties:
      MaxSize: "2"
      MinSize: "1"
      AvailabilityZones: !GetAZs ''
      LaunchTemplate:
        LaunchTemplateId: !Ref MyLT
        Version: !GetAtt MyLT.LatestVersionNumber
    UpdatePolicy:
      AutoScalingRollingUpdate:
        MaxBatchSize: !Ref MyLT   # E1020 in 1.56.0, clean in 1.55.1
```
